lab_1/main.py

```diff
@@ -71,17 +67,19 @@ def get_top_n_words(freq_dict: dict, top_n: int) -> list or None:
     :return: a list of the most common words
     """
 
-    if isinstance(freq_dict, dict) and isinstance(top_n, int):
-        most_common = []
-        chart = sorted(list(freq_dict.values()), reverse=True)
-        chart = chart[:top_n]
-        freq_dict_temp = dict(freq_dict)
-        for i in chart:
-            word = list(freq_dict_temp.keys())[list(freq_dict_temp.values()).index(i)]
-            most_common.append(word)
-            freq_dict_temp.pop(word)
-        return most_common
-    return None
+    if not isinstance(freq_dict, dict) and isinstance(top_n, int):
+        return None
+
+    most_common = []
+    sorted_freqs = sorted(freq_dict.values(), reverse=True)
+    for freq in sorted_freqs:
+        for k, val in freq_dict.items():
+            if val == freq:
+                most_common.append(k)
+
+    most_common = list(dict.fromkeys(most_common))
```

Comment:
can you explain what happens here?

Reply:
Here I append the first found key (= word) from a given dictionary if its value (frequency) matches the value (= frequencies) from the sorted and cut list. The only issue was that the keys were added twice, which is corrected by making a dict and then back to list from the most_common.

Reply:
Hm